src/common/AlarmModal.tsx

```diff
@@ -1,32 +1,82 @@
+import { useNavigate } from 'react-router-dom';
 import styled from 'styled-components';
-import { ALARMLIST } from '../constants/Alarm/alarm';
+import { IcNewAlarm } from '../assets';
+import usePostAlarmRead from '../libs/hooks/Alarm/usePostAlarmRead';
 import { AlarmModalProps } from '../types/Alarm/alarmType';
 
-const AlarmModal = ({ isOpen, handleClose }: AlarmModalProps) => {
-  const { NEWALARMS, READALARMS } = ALARMLIST;
+const AlarmModal = ({
+  isOpen,
+  handleClose,
+  notifications,
+}: AlarmModalProps) => {
+  // notifications가 로드되지 않았을 때 빈 배열을 기본값으로 설정
+  const newAlarms =
+    notifications?.filter((data) => data?.isRead === false) || [];
+  const readAlarms =
+    notifications?.filter((data) => data?.isRead === true) || [];
+  const { mutation } = usePostAlarmRead();
+
+  const navigate = useNavigate();
+
+  // notifications 에서 data 전달
+  const handleAlarmClick = (
+    notificationIds: number,
+    type: string,
+    dataId: number
+  ) => {
+    mutation(notificationIds, {
+      onSuccess: () => {
+        switch (type) {
+          case 'FOLLOW':
+            navigate('/follower');
+            break;
+          case 'CREATED_PUBLIC_ROOM_REQUEST':
+            navigate(`/group/${dataId}/admin`);
+            break;
+          case 'CREATED_PRIVATE_ROOM_JOIN':
+            navigate(`/group/${dataId}/admin`);
+            break;
+          case 'PUBLIC_ROOM_REQUEST':
+            navigate(`/group/${dataId}`, {
+              state: { disabledApply: true },
+            }); // state 를 true로 변경하여 신청하기 버튼 없애기
+            break;
+          case 'PUBLIC_ROOM_APPROVE':
+            navigate(`/group/${dataId}/member`);
+            break;
+          case 'ROOM_STATUS_INACTIVE':
+            navigate('/group/my-page');
+            break;
+        }
+      },
+    });
+  };
```

Comment:
P1: 이 부분 이렇게 수정하고 (커스텀 훅으로 이동)

```suggestion
    const handleAlarmClick = (
    notificationIds: number,
    type: string,
    dataId: number
  ) => {
    mutation({ notificationIds, type, dataId });
  };
```

Reply:
와 훨~씬 더 깔끔해졌네요,, 감사합니다ㅠㅠㅠㅠㅠㅠㅠㅠ